app/scripts/metamask-controller.js

```diff
@@ -510,6 +510,18 @@ export default class MetamaskController extends EventEmitter {
           listener,
         ),
     });
+    // TODO: Remove once `PreferencesController` is migrated to the core repo
+    this.controllerMessenger.registerActionHandler(
+      `PreferencesController:getState`,
+      () => this.preferencesController.store.getState(),
+    );
+    // TODO: Remove once `PreferencesController` is migrated to the core repo
+    this.controllerMessenger.registerInitialEventPayload({
+      eventType: `PreferencesController:stateChange`,
+      getPayload: () => [this.preferencesController.store.getState(), []],
+    });
+    // TODO: Remove once `PreferencesController` is migrated to the core repo
+    this.controllerMessenger.subscribe(`PreferencesController:stateChange`);
```

Comment:
doesn't this end up being a noop since there is no listener/handler?

Reply:
You're absolutely right, thanks for catching this! I refactored the preferences-controller so it uses the controller-messenger pattern and publishes `stateChange` events for the methods token-detection-controller and tokens-controller use (those are currently the only core controllers subscribe to `PreferencesController:stateChange`.

Fixed here: https://github.com/MetaMask/metamask-extension/pull/22928/commits/56eff3fcdffa17679d53a2ff884e13fcde653704